Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs

```diff
@@ -165,21 +152,23 @@ private LoginResult Login(IWebDriver driver, Uri uri, SecureString username, Sec
                 ThinkTime(1000);
             }
 
-            EnterOneTimeCode(driver, mfaSecrectKey);
-
-            ClickStaySignedIn(driver);
-
-            ThinkTime(1000);
+            int attempts = 0;
+            bool entered;
+            do
+            {
+                entered = EnterOneTimeCode(driver, mfaSecrectKey);
+                success = ClickStaySignedIn(driver) || IsUserAlreadyLogged();
+                attempts++;
+            }
+            while (!success && attempts <= Constants.DefaultRetryAttempts); // retry to enter the otc-code, if its fail & it is requested again 
+           
+            if (entered && !success)
+                throw new InvalidOperationException("Somethig got wrong entering the OTC. Please check the MFA-SecrectKey in configuration.");

```

Comment:
retry to enter the otc-code, if its fail & it is requested again, until stay signeg is shown or the main page is loaded